### ISSUE TITLE
std.format %(...%) format does not allow modifiers that take additional arguments

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/phobos#10319]
### CC List

* simen.kjaras

### Description

````markdown
Currently, this is supported:
-----
import std.format;
format("%(%.14f, %)", [ 1.1, 2.2, 3.3 ]);
-----

and so is this:
-----
format("%(%10.3f, %)", [ 1.1, 2.2, 3.3 ]);
-----

However, this is not supported:
-----
format("%(%.*f, %)", double.dig, [ 1.1, 2.2, 3.3 ]);
-----

Nor this:
-----
format("%(%*.3f, %)", 5, [ 1.1, 2.2, 3.3 ]);
-----

Basically, any nested specifier that takes an additional argument is not supported, because of the way nested specifiers are implemented.

It would be nice if this could somehow be made to work.
````

**Comment by dlangBugzillaToGithub** [comment]
#### bugzilla (@WalterBright) commented on 2021-03-27T19:23:32Z



````markdown
*** Issue 20541 has been marked as a duplicate of this issue. ***
````

**Comment by quickfur** [comment]
Main issue is, how would this work? Should `%(%*f, %)` take one extra argument and then the list? Or should it take a list of width/float pairs?

And what about the cast of `%(%s, %s%)`?  How many arguments (list and non-list) should it take, and what should be the semantics?
